data_validation/cli_tools.py

```diff
@@ -1293,8 +1400,41 @@ def get_pre_build_configs(args: Namespace, validate_cmd: str) -> List[Dict]:
             "filter_status": filter_status,
             "trim_string_pks": getattr(args, "trim_string_pks", False),
             "case_insensitive_match": getattr(args, "case_insensitive_match", False),
+            consts.CONFIG_ROW_CONCAT: getattr(args, consts.CONFIG_ROW_CONCAT, None),
+            consts.CONFIG_ROW_HASH: getattr(args, consts.CONFIG_ROW_HASH, None),
             "verbose": args.verbose,
         }
-        pre_build_configs_list.append(pre_build_configs)
+        if (
+            pre_build_configs[consts.CONFIG_ROW_CONCAT]
+            or pre_build_configs[consts.CONFIG_ROW_HASH]
+        ):
+            # Ensure we don't have too many columns for the engines involved.
+            cols = cols_from_arg(
+                pre_build_configs[consts.CONFIG_ROW_HASH]
+                or pre_build_configs[consts.CONFIG_ROW_CONCAT],
+                source_client,
+                table_obj,
+                query_str,
+            )
+            additional_pre_build_configs = _concat_column_count_configs(
+                cols,
+                pre_build_configs,
+                consts.CONFIG_ROW_HASH if args.hash else consts.CONFIG_ROW_CONCAT,
+                _max_concat_columns(
+                    args.max_concat_columns, source_client, target_client
+                ),
+            )
+            if len(additional_pre_build_configs) > 1:
```

Comment:
I didn't want to re-use the name `pre_build_configs` because it was also used as an input into the function. I've changed the new from `additional_pre_build_configs` to `new_pre_build_configs`, I hope that makes it clearer. I still only want to inform the user if there are `> 1` of them, if there's only 1 then we haven't split the validations and there's no need to output a message.

Reply:
Ahh got it, the new name makes easier to understand that, thanks!